louis-smartsearch-frontend/src/components/HighlightedContent.js

```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+
+/**
+ * Component to highlight specific words in a paragraph of content.
+ *
+ * @param {string} content - The paragraph content to be displayed.
+ * @param {string} term - The term to be highlighted in the content.
+ * @returns {JSX.Element} - The rendered component.
+ */
+const HighlightedContent = ({ content, term }) => {
+  // Split the content into individual sentences
+  const sentences = content.split('.').map(sentence => sentence.trim());
```

Comment:
as discuss, extract the code here in a generalized 

function highlight_content(content, query) {
  // split content into sentences
  // split query into words (filtering out <3 characters word
  // split sentences into words
  // normalize words
  // check for matches
  // return list of words and list of index of words that must be highlight
}

then iterate through the list of words, checking if the word needs to be highlighted and returning the correct fragment.

Write tests for highlight_content:

what if the query has multiple word?
what if the query or content is empty?
what if the content is shorter than query?
what if the content has no . ?
what if content or query has punctuation?